massa-execution-worker/src/interface_impl.rs

```diff
@@ -1435,6 +1436,79 @@ impl Interface for InterfaceImpl {
 
         Ok((quotient, remainder))
     }
+
+    fn compare_address_wasmv1(&self, left: &str, right: &str) -> Result<ComparisonResult> {
+        let left = Address::from_str(left)?;
+        let right = Address::from_str(right)?;
+
+        let res = match left.cmp(&right) {
+            std::cmp::Ordering::Less => ComparisonResult::Lower,
+            std::cmp::Ordering::Equal => ComparisonResult::Equal,
+            std::cmp::Ordering::Greater => ComparisonResult::Greater,
+        };
+
+        Ok(res)
+    }
+
+    fn compare_native_amount_wasmv1(
+        &self,
+        left: &NativeAmount,
+        right: &NativeAmount,
+    ) -> Result<ComparisonResult> {
+        let left = amount_from_native_amount(left)?;
+        let right = amount_from_native_amount(right)?;
+
+        let res = match left.cmp(&right) {
+            std::cmp::Ordering::Less => ComparisonResult::Lower,
+            std::cmp::Ordering::Equal => ComparisonResult::Equal,
+            std::cmp::Ordering::Greater => ComparisonResult::Greater,
+        };
+
+        Ok(res)
+    }
+
+    fn compare_native_time_wasmv1(
+        &self,
+        left: &NativeTime,
+        right: &NativeTime,
+    ) -> Result<ComparisonResult> {
+        let left = massa_time_from_native_time(left)?;
+        let right = massa_time_from_native_time(right)?;
+
+        let res = match left.cmp(&right) {
+            std::cmp::Ordering::Less => ComparisonResult::Lower,
+            std::cmp::Ordering::Equal => ComparisonResult::Equal,
+            std::cmp::Ordering::Greater => ComparisonResult::Greater,
+        };
+
+        Ok(res)
+    }
+
+    fn compare_pub_key_wasmv1(&self, left: &str, right: &str) -> Result<ComparisonResult> {
+        let left = PublicKey::from_str(left)?;
+        let right = PublicKey::from_str(right)?;
+
+        let res = match left.cmp(&right) {
+            std::cmp::Ordering::Less => ComparisonResult::Lower,
+            std::cmp::Ordering::Equal => ComparisonResult::Equal,
+            std::cmp::Ordering::Greater => ComparisonResult::Greater,
+        };
+
+        Ok(res)
+    }
+
+    fn compare_sig_wasmv1(&self, left: &str, right: &str) -> Result<ComparisonResult> {
+        let left = Signature::from_str(left)?;
+        let right = Signature::from_str(right)?;
+        todo!()
+        // let res = match left.cmp(&right) {
```

Comment:
@damip 
Our Signature does not have comparison function and the underlying ed25529_dalek::Signature does not implement Ord or PartilOrd.
So I'm wondering how to implement comparison, any tips?

Reply:
Hey indeed I don't think we need order nor comparison on Signature for now !

Reply:
fine, I'm gonna drop this abi :)

Reply:
done